plugins/common/gcp/auth.go

```diff
@@ -0,0 +1,118 @@
+package gcp
```

Comment:
Sure, would you suggest (if we keep this package) we keep this code alongside the cloudrun package code?

Reply:
Kind of. First of all, please check carefully if using `golang.org/x/oauth2/google` is an option instead of really having this code. Second, I tend to making the code in this package an external packages that does not live in telegraf. However, if you have good reasons why option one and two cannot be pursued we should move it to the `cloudrun` plugin directory.